cdk/lib/auth/auth-stack.ts

```diff
@@ -18,20 +19,46 @@ export class AuthStack extends Stack {
   readonly identityPool: CfnIdentityPool;
   readonly userPool: UserPool;
   readonly userPoolClient: UserPoolClient;
+  readonly domain: UserPoolDomain;
 
   constructor(scope: Construct, id: string, props: AuthStackProps) {
     super(scope, id, props);
 
     const { applicationName, logGroupArn, removalPolicyOverride } = props;
 
+    const authMode = this.node.tryGetContext('authMode') as string;
+
     this.userPool = new UserPool(this, 'UserPool', {
       signInCaseSensitive: false,
       removalPolicy: removalPolicyOverride,
     });
 
-    this.userPoolClient = new UserPoolClient(this, 'UserPoolClient', {
-      userPool: this.userPool,
-    });
+    // Cognito domain is required for SSO as an idP
+    if (authMode && authMode === 'sso') {
```

Comment:
You have at least 3 different spots in the CDK code where the code branches to handle SSO. Is it possible to refactor this to have a single conditional and reduce overall cyclomatic complexity? Maybe by creating different classes? Something like `SSOAuthStack extends AuthStack`?